clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,11 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            if (!this.subscriptions.allConsumed().isEmpty() && this.autoCommitEnabled)
```

Comment:
I think we would end up skipping the offset commit anyway since we check for empty prior to sending the offset commit. In fact, we could probably simplify this to just:
```java
if (this.autoCommitEnabled)
  this.commitAsync();
```